src/main/scala/org/broadinstitute/hail/utils/StringEscapeUtils.scala

```diff
@@ -0,0 +1,196 @@
+package org.broadinstitute.hail.utils
+
+import java.util.Locale
+
+import org.broadinstitute.hail.Utils._
+
+object StringEscapeUtils {
+
+  def hex(ch: Char): String = Integer.toHexString(ch).toUpperCase(Locale.ENGLISH)
+
+  def escapeUnprintable(str: String): String = escapeUnprintable(str, new StringBuilder(capacity = str.length * 2))
+
+  def escapeUnprintable(str: String, sb: StringBuilder): String = {
```

Comment:
I'd either make this take `Char` (the only way it is called) and then call it from `escapeString`, or delete it and just use `escapeString` when reporting unprintable characters.